libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -706,6 +716,14 @@ int16_t UARTDriver::read()
         update_rts_line();
     }
 
+#if HAL_LOG_UART_STATS
+    {
+        WITH_SEMAPHORE(_stats.sem);
```

Comment:
I'd be curious how much time grabbing this semaphore for all the drivers that read byte by byte add's up to? I'm just thinking about the companion computer/MAVLink links at near 1 megabaud. It's probably fine, it just seems like it could be a bit of a factor.